packages/compiler-cli/src/transformers/metadata_reader.ts

```diff
@@ -70,7 +70,8 @@ function readMetadataFile(host: MetadataReaderHost, dtsFilePath: string): Module
     return undefined;
   }
   try {
-    const metadataOrMetadatas = JSON.parse(host.readFile(metadataPath));
+    const metadataOrMetadatas =
+        JSON.parse(host.readFile(metadataPath)) as ModuleMetadata[] | undefined;
```

Comment:
I think (to be exact):

```suggestion
        JSON.parse(host.readFile(metadataPath)) as ModuleMetadata | ModuleMetadata[] | undefined;
```

Reply:
And then we can drop the type on the `metadatas` const below.